halo2-base/src/lib.rs

```diff
@@ -143,30 +159,27 @@ impl<F: ScalarField> AsRef<AssignedValue<F>> for AssignedValue<F> {
 
 /// Represents a single thread of an execution trace.
 /// * We keep the naming [Context] for historical reasons.
-#[derive(Clone, Debug)]
+///
+/// [Context] is CPU thread-local.
+#[derive(Clone, Debug, CopyGetters)]
 pub struct Context<F: ScalarField> {
     /// Flag to determine whether only witness generation or proving and verification key generation is being performed.
     /// * If witness gen is performed many operations can be skipped for optimization.
+    #[getset(get_copy = "pub")]
     witness_gen_only: bool,
-
+    /// The challenge phase that this [Context] will map to.
+    #[getset(get_copy = "pub")]
+    phase: usize,
+    /// Identifier for what virtual region this context is in
+    type_id: TypeId,
```

Comment:
Maybe too early to add this because we never read it. No need to remove it though..

Reply:
It's used for example to distinguish between cells in context 0, phase 0 and context 0, phase 1. Also needed to distinguish between context 0, gate advice and context 0, lookup advice